L1Trigger/TrackFindingTracklet/src/TrackletProcessor.cc

```diff
@@ -175,24 +107,58 @@ void TrackletProcessor::addInput(MemoryBase* memory, string input) {
                                  << input;
   }
 
-  if (input == "innervmstubin") {
-    auto* tmp = dynamic_cast<VMStubsTEMemory*>(memory);
-    assert(tmp != nullptr);
-    innervmstubs_.push_back(tmp);
-    setVMPhiBin();
-    return;
-  }
   if (input == "outervmstubin") {
     auto* tmp = dynamic_cast<VMStubsTEMemory*>(memory);
     assert(tmp != nullptr);
-    outervmstubs_.push_back(tmp);
-    setVMPhiBin();
+    outervmstubs_ = tmp;
+    iAllStub_ = tmp->getName()[11] - 'A';
+    if (iSeed_ == 1)
+      iAllStub_ = tmp->getName()[11] - 'I';
+    if (iSeed_ == 6 || iSeed_ == 7) {
+      if (tmp->getName()[11] == 'X')
+        iAllStub_ = 0;
+      if (tmp->getName()[11] == 'Y')
+        iAllStub_ = 1;
+      if (tmp->getName()[11] == 'Z')
+        iAllStub_ = 2;
+      if (tmp->getName()[11] == 'W')
+        iAllStub_ = 3;
+    }
+    buildLUT();  //need iAllStub_ set before building the table //FIXME should be in initiall
+
+    TrackletEngineUnit teunit(&settings_,
+                              nbitsfinephi_,
+                              layerdisk2_,
+                              iSeed_,
+                              nbitsfinephidiff_,
+                              iAllStub_,
+                              pttableinnernew_,
+                              pttableouternew_,
+                              outervmstubs_);
+    teunits_.resize(6, teunit);
+
     return;
   }
+
   if (input == "innerallstubin") {
     auto* tmp = dynamic_cast<AllStubsMemory*>(memory);
     assert(tmp != nullptr);
-    innerallstubs_.push_back(tmp);
+    if (innerallstubs_.size() == 2) {  //FIXME this should be done with better logic with reading the input stubs
+      innerallstubs_.insert(innerallstubs_.begin(), tmp);
+    } else {
+      innerallstubs_.push_back(tmp);
+    }
+
+    //FIXEM should be done once after all inputs are added
+    tedatabuffers_.clear();
+    CircularBuffer<TEData> tedatabuffertmp(5);
+    tedatabuffertmp.reset();
+
+    tedatabuffers_.emplace_back(tuple<CircularBuffer<TEData>, unsigned int, unsigned int, unsigned int, unsigned int>(
```

Comment:
emplace_back just requires the arguments, not the whole constructor, i.e.: `emplace_back(tedatabuffertmp, 0, 0, 0, (innerallstubs_.size() + 1) / 2);`

Reply:
Ok - updated